TWI-11 Get current profile

pm888 opened the pull request.

pm888 committed:
TWI-11 Get current profile

bunin commented on internal/server/server.go:
```diff
@@ -18,7 +18,7 @@ func Server() {
 	r.HandleFunc("/v1/users/login", Serviceuser.LoginUsers).Methods(http.MethodPost)
 	http.Handle("/v1/users/logout", Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.LogoutUser)))
-	http.Handle("/v1/users/{id}", Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.GetCurrentProfile)))
+	r.HandleFunc("/v1/users/{id}", Serviceuser.GetCurrentProfile).Methods(http.MethodGet)
```
why removing Auth middleware, if it's really "current profile" it has to have the authorized user

pm888 committed:
TWI-11 Fix GetCurrentProfile

Add userID from context

pm888 committed:
Merge branch 'main' into TWI-11

bunin commented on openapi.yaml:
```diff
@@ -94,12 +94,12 @@ paths:
-        - name: id
-          in: path
-          description: User ID
+        - name: X-API-KEY
```
this should be in the auth/security section

pm888 committed:
TWI-11 Get current profile

Move: GetCurrentProfile in a separate file
Change: get current profile, *.yaml for Get Current User Profile.
Add: GetCurrentUser struct

bunin commented on internal/server/server.go:
```diff
@@ -17,10 +17,10 @@ func Server() {
-	r.HandleFunc("/v1/logout", func(w http.ResponseWriter, r *http.Request) {
-		Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.LogoutUser)).ServeHTTP(w, r)
+	http.Handle("/v1/users/logout", Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.LogoutUser)))
+	r.HandleFunc("/v1/users/", func(w http.ResponseWriter, r *http.Request) {
```
if I called `GET /v1/users/` I would expect to get a list of users

bunin commented on internal/users/getCurrentUser.go:
```diff
@@ -0,0 +1,47 @@
+		services.ReturnErr(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	user.Followers = subscriptionCount
+	user.Following = followerCount
```
why using additional variables only for these two? why not juts scan into `&user.Followers` and `&user.Following` as you did for the rest of the fields?

pm888 replied:
done

bunin replied:
done where? I don't see any changes here

bunin commented on internal/users/usersStr.go:
```diff
@@ -20,6 +20,15 @@ type Users struct {
 	Following []int
 	Followers []int
 }
+type GetCurrentUser struct {
```
is it a request or a response?

pm888 replied:
done

pm888 replied:
response

bunin replied:
then please call it "*Response"

bunin commented on internal/users/getCurrentUser.go:
```diff
@@ -0,0 +1,47 @@
+	WHERE
+		u.id = $1
+	GROUP BY
+		u.id, u.name, u.birthdate, u.bio, u.location, u.nickname
```
this would need some improvements

pm888 committed:
Merge branch 'main' into TWI-11
